app/models/miq_action.rb

```diff
@@ -795,15 +795,15 @@ def action_cancel_task(action, rec, inputs)
       return
     end
 
-    task_mor = inputs[:ems_event].full_data['info']['task']
+    task_mor = inputs[:source_event].full_data.try(:fetch_path, 'info', 'task')
```

Comment:
Move `inputs[:source_event]` into a separate variable so you do not have to reference the hash each time.

`source_event = inputs[:source_event]`